### PR TITLE
Add autobuild precommit script

### PR DESCRIPTION
**Pull request by brianbolnick**
## Brief context on code (tell us why these changes are necessary)
Adds a pre-commit hook script that will automatically run the build for all of your staged changes. No more needing to `yarn build` 😄  

Now you can go from this:

To this:

## Test plan

There are a few things you should test with this as it should handle a few use cases. It should **not** build any files of uncommitted or unstaged changes. Here are a few scenarios: 

1. Make a change that is build-able (stories work well for this). Stage and commit the change. You should see the build run for that particular file. 

2. Make 2 changes, and only stage one of them. Commit and then verify that only the file you staged has an updated build file.

** Test Cases **

1: Only staged local changes, all have build files
image

2. Staged changes have build files, unstaged changes don't
image

3. Staged new file (Test.story.jsx) includes build files
image

4. Unstaged new file (Test3.story.jsx) does not have build files
image

5.  Deleting a file also removes the build files
Starting with these files:
image

You should then end up with this: 
image

6. Process exits when there is a failed build
With a syntax error or anything that will fail the build, the expected output runs:
image

## Before asking for code review

- [x] I have unit tested behavioral changes
- [x] I have verified test plan works
- [x] I have merged the base branch into this branch
- [x] I have ensured that CI is passing
- [x] I have filled out the "Brief context on code" and "Test plan" sections above
- [x] I have ~rerun the build and~ updated the version in package.json

### CODE REVIEW

File: bin/build-files [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+git stash -k
+echo "RUNNING BUILD...... (\°_°)\ ┬─┬"
+yarn build
```
Comment (andrewjensen) [inline bin/build-files]: If this exits with a non-zero exit status, I think the script will keep going, so weird stuff will happen!

Comment (brianbolnick) [reply bin/build-files]: good catch!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -9,4 +9,4 @@
 - [ ] I have merged the base branch into this branch
 - [ ] I have ensured that CI is passing
 - [ ] I have filled out the "Brief context on code" and "Test plan" sections above
-- [ ] I have rerun the build and updated the version in package.json
+- [ ] I have updated the version in package.json
```
Comment (gseccles) [inline .github/PULL_REQUEST_TEMPLATE.md]: 👏 👏

File: bin/build-files [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	fi
+
+	echo -e "${RED}*****************************"
+	echo -e "BUILD FAILED!!! Try Again...."
+	echo -e "*****************************${NC}"
+
+	exit 1
+fi
+
+#echo -e "${GREEN}BUILD COMPLETE....."
```
Comment (gseccles) [inline bin/build-files]: get out

**Pull request merged**